File: solutions/javascript/balanced-parentheses.js
function isBalanced(s) {
  const stack = [];
  const pairs = { ')':'(', ']':'[', '}':'{' };
  for (let ch of s) {
    if ("([{".includes(ch)) stack.push(ch);
    else if (")]}".includes(ch)) {
      if (!stack.length || stack.pop() !== pairs[ch]) return false;
    }
  }
  return stack.length === 0;
}
console.log(isBalanced("()[{}]"));

File: solutions/java/balanced-parentheses.java
import java.util.*;

public class BalancedParentheses {
    public static boolean isBalanced(String s) {
        Stack<Character> stack = new Stack<>();
        Map<Character,Character> pairs = Map.of(')', '(', ']', '[', '}', '{');
        for (char ch : s.toCharArray()) {
            if ("([{".indexOf(ch) != -1) stack.push(ch);
            else if (")]}".indexOf(ch) != -1) {
                if (stack.isEmpty() || stack.pop() != pairs.get(ch)) return false;
            }
        }
        return stack.isEmpty();
    }
    public static void main(String[] args) {
        System.out.println(isBalanced("()[{}]"));
    }
}